src/common/signal_handling.cpp

```diff
@@ -0,0 +1,32 @@
+#include "common/logging.h"
+#include "signal_handling.h"
+
+// We use signal() here instead of the usual strong recommendation for
+// using sigaction, which apparently is not available for Windows (cf.
+// https://stackoverflow.com/questions/231912/what-is-the-difference-between-sigaction-and-signal).
+
+namespace marian{
+volatile std::sig_atomic_t sigflags_{0};
+volatile std::sig_atomic_t gracefulExitRequested_{0};
+
+bool getSignalFlag(const int sig) {
+  // sig_atomic_t has 32 bits. We don't accommodate signals beyond that.
+  ABORT_IF(sig >= 32, "Signal out of range (must be < 32, is {}).", sig);
```

Comment:
Do we really need this check? I suggest to remove both the comment and the ABORT, since `sig_atomic_t` is, by its name, clearly meant for signals, and therefore defines the valid signal ranges. Marian is not an operating system, we don't need to hedge against every possible invalid argument. You can also remove the check and comment in setSignalFlag

Reply:
sig_atomic_t is meant for signals in the sense that certain operations on it are guaranteed to be atomic even during signal handling. Essentially it is an int type that I'm using here as a bit field, so that we have a generic signal handler that we can install for arbitrary signals (lower than 30). That check is a check against programming errors. Signals can be higher than 30, but this code does not accommodate installing setSignalFlag for such signals.